### ISSUE TITLE
Taxon validation

### ISSUE THREAD

**Issue by helendallas** [kartoza/django-bims#4227]
If the Group contributor (Plecoptera - South Africa) validates a new taxon, then the Group Coordinator (Plecoptera - Global) gets an email, saying the taxon has been validated by Plecoptera-South Africa, it now requires validation to the taxon group Plecoptera- Global. 

BUT, the taxon is already validated at the Global level. It should not be automatically validated  when the Group Contributor validates the taxon. There needs to be a second step whereby the Group Coordinator validates it for the Global Group.  The email  should also be updated to say Plecoptera -Global i.e. the Group coordinator level.  At the moment it is Plecoptera-South Africa.

Initial validation
<img width="1204" alt="Screenshot 2024-09-06 at 11 09 49" src="https://github.com/user-attachments/assets/2db8af78-d7c8-487e-b89b-9c307945e47b">

At Group coordinator, the  taxon is unvalidated (filter=unvalidation), but there is not way for the Group Coordinator to validate - the Approve, Reject option is not visible.

<img width="1217" alt="Screenshot 2024-09-06 at 11 10 34" src="https://github.com/user-attachments/assets/4e06cb11-523e-48df-8758-d4e838a2462a">

**Comment by helendallas** [comment]
please see #3644 . We have only just picked up issue whereby validation is not working correctly.

**Comment by dimasciput** [comment]
@helendallas I have fixed this on staging

**Comment by helendallas** [comment]
@dimasciput I don't think this is sorted yet.

I am the expert in Plecoptera - South Africa.
I added a new taxon, but didn't get a validation noticed.

So I decided to validate it  - and now the taxon is added to Plecoptera - Global but is not added to Plecoptera- South Africa, and no email received.

Please check workflows and emails again.

It should be:

User adds new taxon to Plecoptera- SA
Plecoptera - SA expert - gets email, checks taxon and validates or reject. 
New taxon is then added to Plecoptera- SA

Then Plecoptera - Globel expert gets email saying Plecoptera - SA has validated a new taxon,
then Plecoptera- Global expert has option to Validate or reject.
If validated New taxon is added to Plecoptera - Global

**Comment by helendallas** [comment]
So the above was when I added a synonym from GBIF. let me try again with a taxon not on gif

**Comment by dimasciput** [comment]
hi @helendallas 

Ah, I got the emails, so it's probably just a bug. That's why you're not receiving them. I'll fix it later.

```
User adds new taxon to Plecoptera- SA
Plecoptera - SA expert - gets email, checks taxon and validates or reject.
New taxon is then added to Plecoptera- SA
```

The logic that has been implemented is actually quite different from what you're describing. I believe we've discussed this before, but making changes to this might take some time.

Currently, when a taxon is added to a child taxon group, it isn't automatically added to the parent taxon group.

For example, in a Parent → Child relationship, if a user adds a new taxon to the child group, it will only belong to the child taxon group, even if it is validated. The parent taxon group doesn't include this taxon, it simply checks whether the taxon in the child group has been validated.

What you're asking for is that when a new taxon is added to the child taxon group, it should also be added to the parent group, correct?

This is what we have at the moment

![image](https://github.com/user-attachments/assets/79be0410-f101-40ad-afb4-28a11c11747e)

I think this is what you want? 

![image](https://github.com/user-attachments/assets/2fbfe5ef-938b-4590-b162-863436c7cd72)

There are some challenges with second approach. For instance, if a user rejects the taxon in the parent group but it's still validated in the child group, what happens to the taxon then? We would need to introduce a workflow for the user to propose it again, which adds complexity. The current process is simpler and more straightforward—if a taxon is rejected, it’s rejected across all groups.

There's also some issues when user move the taxon group to other parent.

Could you discuss this with Astrid first? 
Let me fix the email issues for now.

**Comment by helendallas** [comment]
Okay let me discuss with Astrid

**Comment by helendallas** [comment]
Thanks @dimasciput This seems to be working now.

**Comment by schmikloi** [comment]
> Okay let me discuss with Astrid

Hi @helendallas 
Should we discuss this in a short Zoom?
I  mean it was part of the requirements to have multiple levels of data entry, but it does not necessarily have to be connected with a regional-level taxagroup.
The main point here is that someone who is not the highest level expert can suggest a new species (independently from any regional scale/taxalist) and the highest level expert then agrees or disagrees to add this species to the global list.

**Comment by helendallas** [comment]
It seems @dimasciput has added this already. perhaps test the system and see that you are happy

**Comment by helendallas** [comment]
Closing
